Fix #105: Implement time-based separators in agenda view

lambeletjp opened the pull request:
- Add EvenementWithSeparatorCollection to handle separator logic
- Add EventWithSeparator wrapper class with shouldDisplaySeparator() method
- Add Separator class to generate separator labels
- Refactor index.php template to use new collection classes
- Add test data SQL file for time-based separators

lambeletjp committed:
Fix #105: Implement time-based separators in agenda view

- Add EvenementWithSeparatorCollection to handle separator logic
- Add EventWithSeparator wrapper class with shouldDisplaySeparator() method
- Add Separator class to generate separator labels
- Refactor index.php template to use new collection classes
- Add test data SQL file for time-based separators

agilare commented on librairies/EvenementWithSeparatorCollection.php:
```diff
@@ -0,0 +1,85 @@
+        $this->date_next_day = date_lendemain($date_current);
+    }
+
+    public function getIterator(): \Traversable
```
A synthesis of the process and the intention in a DocBlock would be appreciated

agilare reviewed (commented):
Thanks for this proposition, I made some suggestions in the code
Personnaly I would use less classes and variables but it should be also my oldschool procedural style
I did a try with some events and I reveal a problem causing confusion : the hours in separators could mean it the hours of the following event, if we arn't enough attentive

<img width="471" height="1001" alt="Image" src="https://github.com/user-attachments/assets/9f244ecb-c203-4240-8a30-f094cd3c2d46" />

We should find a disambiguation for ex. dès 20h..." instead of only "20h"

<img width="579" height="707" alt="Image" src="https://github.com/user-attachments/assets/e1cf8c4f-2038-4141-a3d0-ad731ee0f404" />

or even a separator for each hour change

lambeletjp committed:
Fix #105: Address PR review - simplify separator classes, hourly separators

- Rename calculateEvenHour to getHour, separators now trigger on each hour change
- Inject Separator objects directly into EventWithSeparator constructor
- Harmonize variable names (is_no_time, event_count, has_shown_no_time_separator)
- Add docblocks to getIterator, getHour, and EventWithSeparator class
- Simplify index.php call to getSeparator()
- Change label to "des Xh" format
- Add composer allow-plugins config for cweagans/composer-patches

agilare commented:
Thank you for your changes according to my review. After an overview I must admit I would write this feature more basically with less (or no) classes and less variables, but again, it's my style.
What is the motivation to iterate the all initial events to get a collection, wrap each event with separator, all in dedicated objects ? It's a way to apply more OOP and patterns like separation of concerns, resusability... ?

lambeletjp commented:
Hi @agilare, thanks for the feedback.

Yes, that is the idea. 
The collection does the loop and picks where to put separators, and the Separator class only holds the data of one separator. Like this, the logic is easy to test on its own, and I can reuse it later (for example on a venue page or favorites).

But I see it can look like too much for a small feature. If you prefer a simpler way with just a loop in the template, I can change it. Let me know.

agilare commented:
Ok, thank you for the explanations, let's go with this design
